### PR TITLE
Add custom log class for CakePHP 4

### LINKED ISSUE #260: Feature Request: Add new class to log data into custom files
Add convenience class method to log data directly into custom files.

So instead of doing this:

```php
use Cake\Log\Log;

Log::config('custom_file', [
    'className' => 'File',
    'path' => LOGS,
    'levels' => ['debug'],
    'file' => 'custom_file.log',
]);

Log::write('debug', 'Data to log into file goes here');
```

With this approach, we have multiple issues:
- Duplicate log data into `$level.log` file and `custom_file.log`
- When we have to write more logs into same file from different parts of the project, we have to copy-paste this code every time.

I think we should have simple API here as below:

```php
use Tools\Log\Log;

Log::write('Data to log into file', 'custom_file');
```

Here we can overcome the issue of duplicating records, also we have reduced code too.

@dereuromark Please let me know your thoughts on this. If you think it's good idea to have this feature in this plugin, I can create a PR to add this thing.

### PR DESCRIPTION
**Pull request by ishan-biztech**
Closes #260

### CODE REVIEW

File: tests/bootstrap.php [hunk 1]
```diff
@@ -85,6 +85,22 @@
 
 Cake\Cache\Cache::setConfig($cache);
 
+Cake\Log\Log::setConfig('debug', [
+	'className' => 'Cake\Log\Engine\FileLog',
+	'path' => LOGS,
+	'file' => 'debug',
+	'levels' => ['notice', 'info', 'debug'],
+	'url' => env('LOG_DEBUG_URL', null),
```
Comment (dereuromark) [inline tests/bootstrap.php:93]: 'scopes' => false,
do we need this here too? does it change anything?

File: src/Utility/FileLog.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+	/**
+	 * Log data into custom file
+	 *
+	 * @param array|string $data Data to store
+	 * @param string|null $filename Filename of log file
+	 * @param bool $traceKey Add trace string key into log data
+	 * @return bool Success
+	 */
+	public static function write($data, $filename = null, $traceKey = false) {
```
Comment (dereuromark) [inline src/Utility/FileLog.php:52]: Do we want to use ?string typehint and bool one here?

Comment (ishan-biztech) [reply src/Utility/FileLog.php]: We are not returning string so only have to typehint `bool`

**Comment by dereuromark** [comment]
tests seem to fail

**Comment by ishan-biztech** [comment]
Fixed

**Pull request merged**